include/nlohmann/json.hpp

```diff
@@ -3522,15 +3592,26 @@ class basic_json
             // fill up array with null values if given idx is outside range
             if (idx >= m_value.array->size())
             {
-                m_value.array->insert(m_value.array->end(),
-                                      idx - m_value.array->size() + 1,
-                                      basic_json());
+#if JSON_DIAGNOSTICS
+                // remember array size before resizing
+                const auto previous_size = m_value.array->size();
+#endif
+
+                m_value.array->resize(idx + 1);
+
+#if JSON_DIAGNOSTICS
+                // set parent for values added above
+                for (auto i = previous_size; i <= idx; ++i)
+                {
+                    m_value.array->operator[](i).m_parent = this;
```

Comment:
I'm torn between leaving this as a direct access to `m_parent` because we're already inside `#if JSON_DIAGNOSTICS ` and using `set_parent(` for consistency.

Reply:
Fixed in f8037660d06a0fbb6b1cde944d436032d04564ae.